xcode/Subconscious/Shared/UIKit/BlockEditor/Common/BlockEditorProfilePicSmView.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  BlockEditorProfilePicSmView.swift
+//  Subconscious (iOS)
+//
+//  Created by Gordon Brander on 9/5/23.
+//
+
+import UIKit
+
+extension BlockEditor {
+    class ProfilePicSmView: UIView, UIViewRenderableProtocol {
+        private var size: CGFloat = 22
+        private var imageView = UIImageView(frame: .zero)
+
+        override init(frame: CGRect) {
+            super.init(frame: frame)
+            
+            imageView.translatesAutoresizingMaskIntoConstraints = false
+            imageView.contentMode = .scaleAspectFill
+            imageView.layer.masksToBounds = false
+            imageView.clipsToBounds = true
+            
+            addSubview(imageView)
+            
+            NSLayoutConstraint.activate([
+                imageView.centerXAnchor.constraint(equalTo: centerXAnchor),
+                imageView.centerYAnchor.constraint(equalTo: centerYAnchor),
+                imageView.widthAnchor.constraint(equalToConstant: size),
+                imageView.heightAnchor.constraint(equalToConstant: size),
+                widthAnchor.constraint(equalToConstant: size),
+                heightAnchor.constraint(equalToConstant: size)
+            ])
+        }
+
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+        
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            // Apply the circle crop.
+            // It's important to run this in `layoutSubviews`, since by this
+            // lifecycle hook the view frame has a height assigned by
+            // autolayout.
+            imageView.layer.cornerRadius = frame.height / 2
+            imageView.backgroundColor = UIColor.secondarySystemFill
+        }
+        
+        func render(_ state: ProfilePicVariant?) {
+            // TODO: implement generative profile pics for UIKit code
```

Comment:
Hm... We could render the SwiftUI view to a `CGImage` _or_ we should be able to rewrite the generative PFP rendering using more primitive drawing operations to directly produce a `CGImage` and render that in SwiftUI too.

The latter seems preferable and hopefully not too hard. We're not using any particularly crazy effects.

Reply:
Rather than going this route, I would eventually like to figure out if embedding SwiftUI in UIKit is reasonable for this kind of thing.

There are complications... the The UIHostingController requires us to manage the lifecycle of each dynamic view, adding and removing from parent controller. But these are probably surmountable.

I was struggling enough with getting constraints working properly in dynamic cells within a UICollectionView that I wanted to hold off on this more complex path, and just get the basics working.